### ISSUE TITLE
[Bug]: Blocked by Cloudflare hCaptcha

### ISSUE THREAD

**Issue by Cubicpath** [Cubicpath/ChatGPT-GUI#7]
### Contact Details

cubicpath@pm.me

### What happened?

ChatGPT's new cloudflare hCaptcha implementation is blocking the use of unofficial apis. When making a request to https://chat.openai.com, you must pass a security check, which involves a JS challenge and a 1x1 pixel  `transparent.gif` resource.

Once you pass the security check, the browser stores a `cf_clearance` cookie which has a 1-year expiration (The token itself should have a 15-45 minute expiration, as documented [here](https://developers.cloudflare.com/fundamentals/security/challenge-passage/#customize-the-challenge-passage)). If this cookie is missing or invalid, you will need to pass the security check again before accessing the website, even if you have a valid session/access token.

### Version

v0.3.3

### What operating system are you using?

Windows

### Relevant log output

```shell
Traceback for UnicodeDecodeError
-----------------------------------
'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte

  File "C:\Projects\Programs\Python\ChatGPT-GUI\src\chatgpt_gui\gui\widgets\conversation_view.py", line 137, in send_message
    app().client.send_message(message, self.conversation)  # type: ignore

  File "C:\Projects\Programs\Python\ChatGPT-GUI\src\chatgpt_gui\network\client\chatgpt.py", line 324, in send_message
    if not (models := self.get_models()):

  File "C:\Projects\Programs\Python\ChatGPT-GUI\src\chatgpt_gui\network\client\chatgpt.py", line 292, in get_models
    return self._get('backend-api/models').json.get('models')

  File "C:\Projects\Programs\Python\ChatGPT-GUI\src\chatgpt_gui\network\manager.py", line 903, in json
    return json_.loads(data.decode(encoding=encoding))
```

**Comment by Cubicpath** [comment]
Cloudflare clearance tokens last ~1 hour, and are tied to your IP + User Agent

**Comment by Cubicpath** [comment]
`cf_clearance` is now fully supported and can be bypassed without user input. Sadly email & password verification doesn't currently work.

This issue is resolved in v0.4.0, which will be released shortly.
